packages/trader/src/AppV2/Components/TradeParameters/RiskManagement/take-profit-and-stop-loss-container.tsx

```diff
@@ -0,0 +1,129 @@
+            stop_loss: stop_loss_current,
+            sl_error_text: sl_error_text_current,
+        } = sl_ref.current;
+
+        const is_tp_empty = take_profit_current === '' && has_take_profit_current;
+        const is_sl_empty = stop_loss_current === '' && has_stop_loss_current;
+        if (is_tp_empty) setTPErrorText(localize('Please enter a take profit amount.'));
+        if (is_sl_empty) setSLErrorText(localize('Please enter a stop loss amount.'));
+        if ((tp_error_text_current && has_take_profit_current) || (sl_error_text_current && has_stop_loss_current))
+            return;
+        if (is_sl_empty || is_tp_empty) return;
+
+        // Show notification, that DC will be disabled if TP or SL is enabled
+        const is_tp_enabled = tp_error_text_current ? false : has_take_profit_current;
+        const is_sl_enabled = sl_error_text_current ? false : has_stop_loss_current;
+        if ((is_tp_enabled || is_sl_enabled) && has_cancellation) {
+            addSnackbar({
+                message: getSnackBarText({
+                    has_cancellation,
+                    has_stop_loss: is_sl_enabled,
+                    has_take_profit: is_tp_enabled,
+                    switching_TP_SL: true,
+                }),
+                hasCloseButton: true,
+                delay: 100,
+            });
+        }
+
+        onChangeMultiple({
+            has_take_profit: has_take_profit_current,
+            take_profit: tp_error_text_current || take_profit_current === '0' ? '' : take_profit_current,
+            has_stop_loss: has_stop_loss_current,
+            stop_loss: sl_error_text_current || stop_loss_current === '0' ? '' : stop_loss_current,
+            ...(is_tp_enabled || is_sl_enabled ? { has_cancellation: false } : {}),
+        });
+
+        closeActionSheet();
+    };
+
+    return (
+        <div className='risk-management__tp-sl__wrapper'>
+            <TakeProfitAndStopLossInput
+                classname='risk-management__tp-sl'
+                has_save_button={false}
+                initial_error_text={tp_error_text}
+                onActionSheetClose={closeActionSheet}
+                parent_subscription_id_ref={tp_subscription_id_ref}
+                parent_ref={tp_ref}
+                parent_is_be_response_received_ref={is_be_response_tp_received_ref}
+                should_wrap_with_actionsheet={false}
+                key='take_profit'
+            />
+            <TakeProfitAndStopLossInput
+                classname='risk-management__tp-sl'
+                has_save_button={false}
+                initial_error_text={sl_error_text}
+                onActionSheetClose={closeActionSheet}
+                parent_subscription_id_ref={sl_subscription_id_ref}
+                parent_ref={sl_ref}
+                parent_is_be_response_received_ref={is_be_response_sl_received_ref}
+                type='stop_loss'
+                should_wrap_with_actionsheet={false}
+                key='stop_loss'
```

Comment:
```suggestion
```
It doesn't look like this key is used ( `key='take_profit'` above as well) inside `TakeProfitAndStopLossInput`. Is it?

Reply:
I can remove, just thought it might cause some problems for react during diffing of Fiber tree.
Maryia, what do you think?

Reply:
Seems that it working without keys, but I didn't do deep testing

Reply:
no worries, it will work fine without keys :)